### PR TITLE
Fixed #28738 -- Added the GeometryDistance function.

### PR DESCRIPTION
**Pull request by franciscouzo**
https://code.djangoproject.com/ticket/28738

### CODE REVIEW

File: django/contrib/gis/db/models/functions.py [hunk 1]
```diff
@@ -299,6 +299,13 @@ def as_mysql(self, compiler, connection, **extra_context):
         return clone.as_sql(compiler, connection, **extra_context)
 
 
+class GeometryDistance(GeoFunc):
+    output_field = FloatField()
```
Comment (charettes) [inline django/contrib/gis/db/models/functions.py]: Define `arity = 2` as well.

File: docs/releases/2.2.txt [hunk 2]
```diff
@@ -62,6 +62,9 @@ Minor features
 * Added SpatiaLite support for the :lookup:`coveredby` and :lookup:`covers`
   lookups.
 
+* Added PostGis support for the
+  :class:`~django.contrib.gis.db.models.functions.GeometryDistance` function.
+
```
Comment (charettes) [inline docs/releases/2.2.txt]: Target 3.0 instead as 2.2 will not receive any new features at this point.

**Comment by franciscouzo** [comment]
> Given this is represented as an operator that accepts two expressions maybe it should be registered as a `Transform` instead?

Similar to `distance_gt`, `distance_gte`, etc? I don't think it makes much sense, it would work the exact same way as those, the benefit on the `<->` operator is when it is used on an `ORDER BY` clause

**Comment by franciscouzo** [comment]
@charettes the jenkin errors don't look related to the changes I made

**Comment by charettes** [comment]
@franciscouzo you are right they are caused by some issues with the CI setup, feel free to ignore them.

**Pull request merged**